### PR TITLE
Adding contextual demo for Gemma 2B

### PR DESCRIPTION
**Pull request by jasonmayes**
Shows how to create a right click interface to bring smarts to any website.

### CODE REVIEW

File: right-click-for-superpowers/index.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE html>
+<html lang="en">
```
Comment (devnook) [inline right-click-for-superpowers/index.html:2]: Please Add SPDX License header (see https://github.com/GoogleChromeLabs/web-ai-demos/blob/main/product-reviews/index.html#L2)

File: right-click-for-superpowers/index.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0" />
+    <meta name="author" content="Jason Mayes" />
+    <title>Gemma 2B LLM Right Click Superpowers</title>
+    <link rel="stylesheet" href="style.css" type="text/css" />
```
Comment (devnook) [inline right-click-for-superpowers/index.html:8]: Please use relative urls if possible, to enable the demo to run on chrome.dev (they run on a deep url like chrome.dev/my-demo)

File: right-click-for-superpowers/index.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    </div>
+      
+    <div id="preloader" class="animate__animated animate__fadeIn">
+      <h2  class="animate__animated animate__pulse animate__infinite">Please wait. Loading 1.3GB Model file.</h2>
+    </div>
+    
+    <footer>
+      Created by <a href="https://www.linkedin.com/in/WebAI" target="_blank">Jason Mayes</a>, 2024. Note: This is not an official Google product.
+    </footer>
+    <script type="text/javascript" src="script.js"></script>
```
Comment (devnook) [inline right-click-for-superpowers/index.html:35]: Pls relative url

File: right-click-for-superpowers/script.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
+import {FilesetResolver, LlmInference} from 'https://cdn.jsdelivr.net/npm/@mediapipe/tasks-genai';
```
Comment (devnook) [inline right-click-for-superpowers/script.js:1]: Please add license header to all files containing code. E.g. https://github.com/GoogleChromeLabs/web-ai-demos/blob/main/product-reviews/src/.consts.js#L1

File: right-click-for-superpowers/script.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+const MODEL_FILE_NAME = 'https://storage.googleapis.com/jmstore/kaggleweb/grader/g-2b-it-gpu-int4.bin';
```
Comment (devnook) [inline right-click-for-superpowers/script.js:3]: I'm not sure we want to check in this url. Maybe leave a placeholder for the user to fill in, like in this demo: https://github.com/GoogleChromeLabs/web-ai-demos/blob/main/product-reviews/.env.template

You can add a note in Readme that developer needs to provide path/url to the downloaded model.

File: right-click-for-superpowers/script.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+let promptText = '';
+let llmInference = undefined;
```
Comment (devnook) [inline right-click-for-superpowers/script.js:6]: let llmInference; will be enough

File: right-click-for-superpowers/script.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+let genText = '';
+let lastSelectedText = '';
+let menu = document.getElementById('menu');
+let menuItems = document.querySelectorAll('.item');
+let sidePanel = document.getElementById('sidePanel');
+let preloader = document.getElementById('preloader');
```
Comment (devnook) [inline right-click-for-superpowers/script.js:12]: These all can be const instead of let

File: right-click-for-superpowers/script.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+for (let i = 0; i < menuItems.length; i++) {
+  menuItems[i].addEventListener('click', handleMenuClick);
+}
+
+document.body.addEventListener('contextmenu', function (e) {
+  e.preventDefault();
+  menu.setAttribute('style', 'left: ' + e.clientX + 'px; top: ' + e.clientY + 'px;');
+  menu.classList.add('enabled');
+  lastSelectedText = getSelectionText();
+  console.log(lastSelectedText);
```
Comment (devnook) [inline right-click-for-superpowers/script.js:23]: Pls remove console.logs

File: right-click-for-superpowers/script.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+      break;
+    case 'Translate to German':
+      promptText = 'Translate the following sentence to German: "jasonholder"';
+      break;
+    default:
+      promptText = '';
+  }
+  
+  if(llmInference !== undefined) {
+    let promptFinal = promptText.replace('jasonholder', lastSelectedText).split('\n').join(' ');
```
Comment (devnook) [inline right-click-for-superpowers/script.js:56]: const

File: right-click-for-superpowers/style.css [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+.removed {
+  display: none;
+}
+
+@charset "UTF-8";/*!
+ * animate.css - https://animate.style/
+ * Version - 4.1.1
+ * Licensed under the MIT license - http://opensource.org/licenses/MIT
```
Comment (devnook) [inline right-click-for-superpowers/style.css:110]: We cannot check in 3rd party code. Let's link to the cdn instead? E.g
```
<link href="
https://cdn.jsdelivr.net/npm/animate.css@4.1.1/animate.min.css
" rel="stylesheet">
```

**Review by devnook — COMMENTED** [review]
This is great, Jason! I left some initial comments below.

File: right-click-for-superpowers/index.html (hunk 2 above)
Comment (jasonmayes) [reply right-click-for-superpowers/index.html:8]: This is a relative URL. Would work from whatever folder it is served from so long as the 3 files are in the same folder.

File: right-click-for-superpowers/index.html (hunk 3 above)
Comment (jasonmayes) [reply right-click-for-superpowers/index.html:35]: Same here, this is a relative URL

File: right-click-for-superpowers/index.html (hunk 1 above)
Comment (jasonmayes) [reply right-click-for-superpowers/index.html:6]: Done

File: right-click-for-superpowers/script.js (hunk 4 above)
Comment (jasonmayes) [reply right-click-for-superpowers/script.js:6]: Done

File: right-click-for-superpowers/script.js (hunk 5 above)
Comment (jasonmayes) [reply right-click-for-superpowers/script.js:8]: Done

File: right-click-for-superpowers/script.js (hunk 6 above)
Comment (jasonmayes) [reply right-click-for-superpowers/script.js:11]: Updated though I like to be explicit so anyone reading knows its state incase they dont know the default value is that.

File: right-click-for-superpowers/script.js (hunk 7 above)
Comment (jasonmayes) [reply right-click-for-superpowers/script.js:17]: Updated

File: right-click-for-superpowers/script.js (hunk 8 above)
Comment (jasonmayes) [reply right-click-for-superpowers/script.js:28]: Done

File: right-click-for-superpowers/script.js (hunk 9 above)
Comment (jasonmayes) [reply right-click-for-superpowers/script.js:61]: Updated

File: right-click-for-superpowers/style.css (hunk 10 above)
Comment (jasonmayes) [reply right-click-for-superpowers/style.css]: Done

**Comment by jasonmayes** [comment]
Thanks for the comments. Have addressed them.

**Review by devnook — APPROVED** [review]

**Pull request merged**